### PR TITLE
Add event ids to all log calls

### PR DESCRIPTION
**Pull request by pakrym**
https://github.com/aspnet/DataProtection/issues/94

@victorhurdugaci @moozzyk @anpete

### CODE REVIEW

File: src/Microsoft.AspNet.DataProtection/LoggingExtensions.cs [hunk 1]
```diff
@@ -13,6 +14,264 @@ namespace Microsoft.Extensions.Logging
...
+
+        private static Action<ILogger, Guid, Exception> _triedToProcessRevocationOfKeyButNoSuchKeyWasFound;
+
+        private static Action<ILogger, Guid, Exception> _foundKey;
+
+        private static Action<ILogger, DateTimeOffset, Exception> _foundRevocationOfAllKeysCreatedPriorTo;
+
+        private static Action<ILogger, Guid, Exception> _foundRevocationOfKey;
+
+        private static Action<ILogger, XElement, Exception> _ExceptionWhileProcessingRevocationElement;
```
Comment (Eilon) [inline src/Microsoft.AspNet.DataProtection/LoggingExtensions.cs]: Oh no, field starts with capital `E` :smile:

File: src/Microsoft.AspNet.DataProtection/LoggingExtensions.cs [hunk 2]
```diff
@@ -13,6 +14,264 @@ namespace Microsoft.Extensions.Logging
     /// </summary>
     internal static class LoggingExtensions
     {
+        private static Action<ILogger, Guid, DateTimeOffset, Exception> _usingFallbackKeyWithExpirationAsDefaultKey;
```
Comment (victorhurdugaci) [inline src/Microsoft.AspNet.DataProtection/LoggingExtensions.cs]: Unrelated, we should use a generator (T4?) to generate these classes. It's a pain to maintain them

File: src/Microsoft.AspNet.DataProtection/LoggingExtensions.cs [hunk 3]
```diff
@@ -69,44 +328,350 @@ private static bool IsLogLevelEnabledCore(ILogger logger, LogLevel level)
...
 
-        public static void LogDebugF(this ILogger logger, Exception error, FormattableString message)
+        public static void WritingDataToFile(this ILogger logger, string finalFilename)
         {
-            logger.LogDebug(new FormattedLogValues(message.Format, message.GetArguments()), error);
+            _writingDataToFile(logger, finalFilename, null);
         }
 
-        public static void LogError(this ILogger logger, Exception error, string message)
+        public static void ReadingDataFromRegistryKeyValue(this ILogger logger, Win32.RegistryKey regKey, string valueName)
```
Comment (victorhurdugaci) [inline src/Microsoft.AspNet.DataProtection/LoggingExtensions.cs]: Use namespace?

**Comment by victorhurdugaci** [comment]
Looks good :shipit: assuming the CI passes

**Comment by pakrym** [comment]
:up: :date:
Requires https://github.com/aspnet/Logging/pull/296

File: src/Microsoft.AspNet.DataProtection/LoggingExtensions.cs [hunk 4]
```diff
@@ -69,44 +421,294 @@ private static bool IsLogLevelEnabledCore(ILogger logger, LogLevel level)
             return (logger != null && logger.IsEnabled(level));
         }
 
-        public static void LogDebugF(this ILogger logger, FormattableString message)
+        public static void UsingFallbackKeyWithExpirationAsDefaultKey(this ILogger logger, Guid keyId, DateTimeOffset expirationDate)
+        {
+            _usingFallbackKeyWithExpirationAsDefaultKey(logger, keyId, expirationDate, null);
```
Comment (Eilon) [inline src/Microsoft.AspNet.DataProtection/LoggingExtensions.cs]: Named param for the `null` at the end of these?

**Comment by Eilon** [comment]
Just small comments, then :shipit:

File: src/Microsoft.AspNet.DataProtection/LoggingExtensions.cs (hunk 4 above)
Comment (pakrym) [inline src/Microsoft.AspNet.DataProtection/LoggingExtensions.cs]: Nope, its Action<?> so param name would be argN

Comment (Eilon) [inline src/Microsoft.AspNet.DataProtection/LoggingExtensions.cs]: Oh right, I keep forgetting that.

Comment (moozzyk) [inline src/Microsoft.AspNet.DataProtection/LoggingExtensions.cs]: good ol' inline comment `/* parameterX */ null`

**Comment by GrabYourPitchforks** [comment]
You're going to have to rerun perf tests against this library. The code was originally written in such a way that gathering the data to be written to the log was deferred until it was known for certain that there was a logger listening. For instance, the expensive `JoinPurposesForLog` operation was never called unless the was a logger present and it had the *debug* level enabled. This method is now called for all calls to Protect and Unprotect, even if no logger is present. (There are other instances of expensive operations, but this is the most obvious.)

/cc @blowdart

**Comment by pakrym** [comment]
@ GrabYourPitchforks I left checks in that places

**Comment by GrabYourPitchforks** [comment]
@pakrym great :)

**Pull request merged**